DevTools/SourceMaps/explainer.md

```diff
@@ -0,0 +1,244 @@
+* This document status: **Active**
+* Expected venue: Ad-hoc discussions, to be expanded if there is interest in the community
+* **Current version: this document**
+
+### Revising the standard
+
+We want to recognize the previous standard and the contributions the former authors have made as part of this implementation. We are taking this avenue of publishing in the Microsoft Edge Explainers repository because the former standard has not been updated in 8 years, and the Mozilla source maps discussion list is not even available in archives.
+
+## Introduction
+
+The [previous version of the Source Maps specification](https://sourcemaps.info/spec.html) is reasonably thorough in supporting live debugging. However, one shortcoming that the specification lacks is that it isn't obvious how to decode the names of functions on the call stack. Several implementations (including Node.js and a library [stacktrace-gps](https://github.com/stacktracejs/stacktrace-gps)) use name guessing, and [pasta-sourcemaps](https://github.com/bloomberg/pasta-sourcemaps) uses precompilation and a custom extension to reconstruct a stack trace.
+
+## Goals
+
+ * Improve post-hoc debugging capabilities
+ * Preserve compatibility with existing tooling
+
+## Non-Goals
+
+Nothing specific at this time.
+
+## Use Cases
+
+*The following was taken from [an issue opened on the TypeScript repo](https://github.com/microsoft/TypeScript/issues/46695):*
+
+Suppose I receive the following information from my application:
+
+```
+TypeError: Cannot read properties of undefined (reading 'x')
+    at re.draw (bundle.js:1:16372)
+    at re.drawLayer (bundle.js:1:14170)
+    at be.draw (bundle.js:1:74592)
+    at Te.render (bundle.js:1:114230)
+    at Te.reflowCanvas (bundle.js:1:113897)
+    at HTMLDocument.anonymous (bundle.js:1:135849)
+```
+
+The source map will resolve the first stack frame to `draw(src, sprite, x, y, frame)`, and will correctly point out that the failure here is that the undefined value is actually `frame`. However, *that is not the name of the function*. The function's name is `draw`, which is a member of the `Render` class.
+
+If I simply apply the value of the `names` array to the function, decoding the stack trace is not particularly useful. It would look something like this:
+
+```
+TypeError: Cannot read properties of undefined (reading 'x')
+    at frame (file1.ts:302:7)
+    at draw (file1.ts:144:5)
+    at Render (file2.ts:95:5)
+    at drawArray (file3.ts:178:5)
+    at render (file3.ts:155:5)
+    at game (file4.ts:39:5)
+```
+
+Using the source map to navigate the source code by hand, I can reconstruct the original call stack:

```

Comment:
```suggestion
The source line information here is improved and allows me to more easily find the source files and lines, but the function names are modified and now incorrect. Using the source map to navigate the source code by hand, I can reconstruct the original call stack including the class full names and anonymous functions:

```

I had to jump back and forth between these three stacks a few times to fully grock what was shifting (both good and bad) between them, so trying to add a touch more detail explaining the differences.